tests/tasks/assert_certificate_parameters.yml

```diff
@@ -0,0 +1,35 @@
+---
+- name: Verify certificate exists in the correct location
+  file:
+    path: "{{ cert['path'] }}"
+    state: file
+  register: result
+
+- name: Verify key exists in the correct location
+  file:
+    path: "{{ cert['key_path'] }}"
+    state: file
+  register: result
+
+- name: Verify certificate subject
+  shell: >-
+    openssl x509 -in {{ cert['path'] }} -noout -subject |
```

Comment:
Maybe this test could check for the existence of `cert['path']` as well as testing the contents of the cert.  That is, I think `openssl x509 -in file` will fail if `file` does not exist.  Unless you need to distinguish between "file does not exist" and "file contents are not correct"

Reply:
The idea is that if the file does not exist the error would happen in some assertion before we get here.